# --- ROCK-COPYRIGHT-NOTE-BEGIN ---
# 
# This copyright note is auto-generated by ./scripts/Create-CopyPatch.
# Please add additional copyright information _after_ the line containing
# the ROCK-COPYRIGHT-NOTE-END tag. Otherwise it might get removed by
# the ./scripts/Create-CopyPatch script. Do not edit this copyright text!
# 
# ROCK Linux: rock-src/package/base/dietlibc/adds_vasprintf_c.patch
# ROCK Linux is Copyright (C) 1998 - 2003 Clifford Wolf
# 
# This program is free software; you can redistribute it and/or modify
# it under the terms of the GNU General Public License as published by
# the Free Software Foundation; either version 2 of the License, or
# (at your option) any later version. A copy of the GNU General Public
# License can be found at Documentation/COPYING.
# 
# Many people helped and are helping developing ROCK Linux. Please
# have a look at http://www.rocklinux.org/ and the Documentation/TEAM
# file for details.
# 
# --- ROCK-COPYRIGHT-NOTE-END ---

diff -u -rduN dietlibc-0.22.orig/libugly/vasprintf.c dietlibc-0.22/libugly/vasprintf.c
--- dietlibc-0.22.orig/libugly/vasprintf.c	1969-12-31 21:00:00.000000000 -0300
+++ dietlibc-0.22/libugly/vasprintf.c	2003-11-15 00:33:36.000000000 -0300
@@ -0,0 +1,20 @@
+#define _GNU_SOURCE
+#include <stdarg.h>
+#include <sys/types.h>
+#include <stdlib.h>
+#include <stdio.h>
+#include "dietwarning.h"
+
+int vasprintf(char **s, const char *format, va_list ap)
+{
+  int n;
+  va_list arg_ptr;
+  va_copy(arg_ptr,ap);
+  n=vsnprintf(0,1000000,format,arg_ptr);
+  va_end(arg_ptr);
+  if ((*s=malloc(n+1))) {
+    n=vsnprintf(*s,n+1,format,ap);
+    return n;
+  }
+  return -1;
+}
